File: studies/movies/ggplot2/line_colored.R
library(tidyverse)
library(lubridate)

df <- read_csv("data/movies.csv") |>
  mutate(`Release Date` = year(`Release Date`))
ggplot(df) +
  geom_line(aes(
    x = `Release Date`,
    y = `Worldwide Gross`,
    color = `MPAA Rating`
  ), stat = "summary", fun = "sum")

File: galleries/movies/seaborn_obj/line_colored.py
import seaborn.objects as so
import pandas as pd

df = pd.read_csv("data/movies.csv", parse_dates=["Release Date"])
df["Release Date"] = df["Release Date"].dt.year

p = so.Plot(df, x="Release Date", y="Worldwide Gross", color="MPAA Rating").add(
    so.Line(), so.Agg("sum")
)
